src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchViewModel.cs

```diff
@@ -1564,12 +1572,20 @@ public void DisableSearchTextBox()
         /// <summary>
         /// Clear after closing down
         /// </summary>
-        internal void Close()
+        internal void PackageManagerViewClose()
+        {
+            SearchAndUpdateResults(String.Empty); // reset the search text property
+            InitialResultsLoaded = false;
+            TimedOut = false;
+        }
+
+        /// <summary>
+        /// Remove PackageManagerSearchViewModel resources
+        /// </summary>
+        internal void Dispose()
         {
-            TimedOut = false;   // reset the timedout screen 
-            InitialResultsLoaded = false;   // reset the loading screen settings
-            RequestShowFileDialog -= OnRequestShowFileDialog;
-            nonHostFilter.ForEach(f => f.PropertyChanged -= filter_PropertyChanged);
+             nonHostFilter?.ForEach(f => f.PropertyChanged -= filter_PropertyChanged);
+             aTimer.Elapsed -= new ElapsedEventHandler(OnTimedEvent);
```

Comment:
OnTimedEvent is already defined as a `ElapsedEventHandler`
you can just do `aTimer.Elapsed += OnTimedEvent` and `aTimer.Elapsed -= OnTimedEvent`